### ISSUE TITLE
Allow .module uploads on Module installation

### ISSUE THREAD

**Issue by philippreiner** [ryancramerdesign/ProcessWire#849]
It should be possible to upload a .module file when installing modules by file upload. Currently, it only supports .zip files. 

Example error:
```
TextformatterSoundcloudEmbed.module - Invalid file extension, please use one of: zip, zip
```

**Comment by mike-anthony** [comment]
I'm going to **+1** this.

**Comment by ryancramerdesign** [comment]
Having the module in a ZIP file provides a layer of protection should some corruption happen to the source file. If the file becomes corrupted, it won't unzip, and PW won't install it. Whereas if a ".module" file becomes corrupt, there's no extra layer to indicate that. Lets say you uploaded an autoload module: it would take down your site/application without an easy way to recover short of going directly to the file system and deleting the file. Keeping it in a ZIP also provides a little more portability with reasonable assurance that the file is consistent with what the developer of the module created, and not modified by encodings of different platforms and ftp modes, etc.

**Comment by mike-anthony** [comment]
Yeah, I should have thought of that too. The idea just sounds pretty decent. So I do agree with your points. Now that I think about it, I think consistency is actually a big thing, besides any issues that may arise out of this.

Also, it's only an extra step to zip it up.

**Comment by owzim** [comment]
Good points @ryancramerdesign, totally agree.
